src/go/wsl-helper/pkg/dnsmonitor/run.go

```diff
@@ -0,0 +1,121 @@
+//go:build windows
+// +build windows
+
+/*
+Copyright © 2022 SUSE LLC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package dnsmonitor
+
+import (
+	"bytes"
+	"crypto/md5"
+	"fmt"
+	"net/netip"
+	"os"
+	"sort"
+
+	"golang.org/x/sys/windows"
+	"golang.zx2c4.com/wireguard/windows/tunnel/winipcfg"
+)
+
+func Run(path string) error {
+	addrs, err := winipcfg.GetAdaptersAddresses(windows.AF_UNSPEC, winipcfg.GAAFlagIncludeAll)
+	if err != nil {
+		return err
+	}
+	var activeInterfaces []*winipcfg.IPAdapterAddresses
+	for _, addr := range addrs {
+		if addr.OperStatus == winipcfg.IfOperStatusUp && addr.FirstDNSServerAddress != nil {
```

Comment:
This seems like it would do the wrong thing if the interface ends up not having an IPv4 address?

Reply:
The FirstDNSServerAddress is a socket address of type any (both IPV4 and IPV6).

Reply:
We should make sure that we don't pick an interface that is IPv6 only.

Reply:
@jandubois sure, the sort will take care of that, since we are sorting by IPV4Metric there is a level of assurance that IPV4 will exist.